sparkbench/ml/pom.xml

```diff
@@ -54,4 +54,194 @@
+              </excludes>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+      <activation>
+        <property>
+          <name>spark</name>
+          <value>2.1</value>
+        </property>
+      </activation>
+    </profile>
+
+    <profile>
+      <id>spark2.2</id>
+      <properties>
+        <spark.version>2.2.0</spark.version>
+        <spark.bin.version>2.2</spark.bin.version>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>net.alchim31.maven</groupId>
+            <artifactId>scala-maven-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>com/intel/sparkbench/ml/MLDenseKMeans.scala</exclude>
+              </excludes>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+      <activation>
+        <property>
+          <name>spark</name>
+          <value>2.2</value>
+        </property>
+      </activation>
+    </profile>
+
+    <profile>
+      <id>spark2.3</id>
+      <properties>
+        <spark.version>2.3.0</spark.version>
+        <spark.bin.version>2.3</spark.bin.version>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>net.alchim31.maven</groupId>
+            <artifactId>scala-maven-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>com/intel/sparkbench/ml/MLDenseKMeans.scala</exclude>
+              </excludes>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+      <activation>
+        <property>
+          <name>spark</name>
+          <value>2.3</value>
+        </property>
+      </activation>
+    </profile>
+
+    <profile>
+      <id>spark2.4</id>
+      <properties>
+        <spark.version>2.4.0</spark.version>
+        <spark.bin.version>2.4</spark.bin.version>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>net.alchim31.maven</groupId>
+            <artifactId>scala-maven-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>com/intel/sparkbench/ml/DenseKMeans.scala</exclude>
+              </excludes>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+      <activation>
+        <property>
+          <name>spark</name>
+          <value>2.4</value>
+        </property>
+      </activation>
+    </profile>
+
+    <profile>
+      <id>spark3.0</id>
+      <properties>
+        <spark.version>3.0.0</spark.version>
+        <spark.bin.version>3.0</spark.bin.version>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>net.alchim31.maven</groupId>
+            <artifactId>scala-maven-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>com/intel/sparkbench/ml/DenseKMeans.scala</exclude>
```

Comment:
Is this really necessary? I think at least this need to be clearly documented, mentioning when RDD KMeans/Dataset KMeans is leveraged.

Another option is we just remove the mllib KMeans from this release, only keeping the ml version. This approach makes more sense to me. cc @carsonwang

Reply:
yes, it is fine to remove Spark 2.2 2.3 support and only keep the ML version for Spark 2.4 and 3.x.

Reply:
OK. it's simpler. I will revert the last commit.